provider/caleta/model_mapper.go

```diff
@@ -46,133 +46,141 @@ func (s *ProviderService) refreshSessionMapper(ctx context.Context, token, reque
 	}
 }
 
-func (s *ProviderService) betTransactionMapper(ctx context.Context, r WalletbetJSONRequestBody) pam.AddTransactionRequestMapper {
-	return func(ar pam.AmountRounder) (context.Context, *pam.AddTransactionRequest, error) {
-		amt := toPamAmount(&r.Amount)
+func valueOrNow(t *time.Time) time.Time {
+	if t == nil {
+		return time.Now().UTC()
+	} else {
+		return *t
+	}
+}
+
+func betTransactionMapper(ctx context.Context, r *WalletbetRequestObject) pam.AddTransactionRequestMapper {
+	return func(_ pam.AmountRounder) (context.Context, *pam.AddTransactionRequest, error) {
+		amt := toPamAmount(r.Body.Amount)
 		return ctx, &pam.AddTransactionRequest{
-			PlayerID: r.SupplierUser,
+			PlayerID: r.Body.SupplierUser,
```

Comment:
Relying on schema validation to guard against nil for the `body` field. Is that enough?

Reply:
Should be 🤞